src/components/outline-nav/components/outline-link/index.tsx

```diff
@@ -0,0 +1,21 @@
+import classNames from 'classnames'
+import type { OutlineLinkProps } from '../../types'
+import s from './outline-link.module.css'
+
+/**
+ * Renders an individual outline nav link.
+ */
+function OutlineLink({ title, url, isActive }: OutlineLinkProps) {
+	return (
+		<a
+			className={classNames(s.root, { [s.isActive]: isActive })}
+			href={url}
+			/* TODO: confirm specifics for data-heap-track here, if we want it */
```

Comment:
@BrandonRomano Do you know if we need any specific `data-heap-track` attributes for `OutlineNav` links? Here I've kept something analogous to `table-of-contents`. Let me know what you think, wanted to confirm this as I wasn't 100% sure 🙇

Reply:
This is a good call-out, but I am not sure of the answer.

I would bubble this question up to the team to see if anyone has any idea.

Reply:
cc @anthonyvetter ☝🏻

Reply:
What was the attribute value for table of contents? We may want to keep it consistent across all sidecars. Perhaps we could pass in the attribute as a prop 🤔

Reply:
In `TableOfContents`, [it was `data-heap-track={"toc-list-item-index-" + index}`](https://github.com/hashicorp/dev-portal/blob/dd70254ff2ff1cb97080e6b0b78794f641db1b92/src/components/table-of-contents/index.tsx#L89). I wasn't sure about the `index` part there, and it'll perhaps be a bit different now with nesting. I'm not familiar with how these analytics are used, so would be curious as well to get input.

Re: passing in as a prop, I feel like a consistent prop value might be useful in some way (eg tracking interactions with this component at a global level might give some insight?), and feels like generally the prop value + URL will let us dive into more narrow contexts if we need to. Again though would defer to analytics folks - if there are specific contexts where we want to render different analytics values, then maybe we take a more props-based approach 👍

Reply:
@kendallstrautman Thanks for the suggestion re: attribute-as-prop, took a second for me to catch up but it seems to work way better 💯 I've updated as follows:
- The `OutlineLinkItem` interface now accepts an optional `dataHeapTrack` string attribute, which is passed to `data-heap-track`. (Note this also applies to `OutlineLinkWithNesting`). `OutlineNav` has no opinion on what `dataHeapTrack` should be.
- In `OutlineNavFromHeadings`, which is what we'll be swapping for existing `TableOfContents` use cases, we tack on `dataHeapTrack` properties when we map the flat array `headings` to outline items. This achieves parity with the previous `data-heap-track` attributes in `TableOfContents.

**Note**: this PR does _not_ yet address what `data-heap-track` attributes will be used for nested cases (it is impossible to create nesting with `OutlineNavFromHeadings`). Heap tracking on nested items will be handled in a future PR where we add nesting (#1667).